compiler/src/iree/compiler/Codegen/Interfaces/TensorMaskingOpInterface.cpp

```diff
@@ -0,0 +1,271 @@
+// Copyright 2025 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Codegen/Interfaces/TensorMaskingOpInterface.h"
+#include "iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.h"
+
+#include "mlir/Analysis/SliceAnalysis.h"
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/Dialect/Linalg/Transforms//Transforms.h"
+#include "mlir/Dialect/UB/IR/UBOps.h"
+#include "mlir/IR/BuiltinTypes.h"
+
+namespace mlir::iree_compiler {
+
+#include "iree/compiler/Codegen/Interfaces/TensorMaskingOpInterface.cpp.inc" // IWYU pragma: keep
+
+namespace {
+
+static bool isIterationCarriedArg(OpOperand &arg, linalg::GenericOp op) {
+  auto bbArg = dyn_cast<BlockArgument>(arg.get());
+  if (!bbArg) {
+    return false;
+  }
+  OpOperand *inputOperand = op.getMatchingOpOperand(bbArg);
+  return op.isDpsInit(inputOperand);
+}
+
+static FailureOr<linalg::PadTilingInterfaceResult>
+maskTilingInterfaceOp(OpBuilder &b, Operation *op,
+                      ArrayRef<OpFoldResult> padMultiples) {
+  auto poison = ub::PoisonAttr::get(b.getContext());
+  SmallVector<Attribute> padValues(op->getNumOperands(), poison);
+  TilingInterface tilingInterfaceOp = cast<TilingInterface>(op);
+  // Set options.
+  linalg::PadTilingInterfaceOptions options =
+      linalg::PadTilingInterfaceOptions()
+          .setPaddingSizes(padMultiples)
+          .setPaddingValues(padValues)
+          .setPadToMultipleOf(true);
+
+  FailureOr<linalg::PadTilingInterfaceResult> result =
+      linalg::rewriteAsPaddedOp(b, tilingInterfaceOp, options);
+  return result;
+}
+
+static SmallVector<OpFoldResult>
+getBoundsToGuard(OpBuilder &b, Operation *unpaddedOp, Operation *paddedOp) {
+  OpBuilder::InsertionGuard guard(b);
+  auto unpaddedIface = cast<TilingInterface>(unpaddedOp);
+  auto paddedIface = cast<TilingInterface>(paddedOp);
+  b.setInsertionPoint(unpaddedOp);
+  SmallVector<Range> oldItSpace = unpaddedIface.getIterationDomain(b);
+  b.setInsertionPoint(paddedOp);
+  SmallVector<Range> newItSpace = paddedIface.getIterationDomain(b);
+  SmallVector<OpFoldResult> oldDims =
+      llvm::map_to_vector(oldItSpace, [&](Range r) { return r.size; });
+  SmallVector<OpFoldResult> newDims =
+      llvm::map_to_vector(newItSpace, [&](Range r) { return r.size; });
```

Comment:
These don't capture anything
```suggestion
      llvm::map_to_vector(oldItSpace, [](Range r) { return r.size; });
  SmallVector<OpFoldResult> newDims =
      llvm::map_to_vector(newItSpace, [](Range r) { return r.size; });
```